### PR TITLE
[SDK] Fix function name validation

### PR DESCRIPTION
**Pull request by adamdelman**
[SDK] Fix function name validation when calling the following functions:
|        1. set_function()
|        2. create_model_monitoring_function()
|        3. set_model_monitoring_function

### CODE REVIEW

**Review by moranbental — CHANGES_REQUESTED** [review]
Please add the Jira ticket to the PR description

File: mlrun/projects/project.py [hunk 1]
```diff
@@ -2424,9 +2424,18 @@ def _instantiate_function(
...
 
         if isinstance(func, str):
-            # in hub or db functions name defaults to the function name
-            if not name and not (func.startswith("db://") or func.startswith("hub://")):
-                raise ValueError("Function name must be specified")
+            # In hub or db functions name defaults to the function name
+            # If name is specified, we try to create a new function with this name.
+            if not name:
+                if func and not (func.startswith("db://") or func.startswith("hub://")):
+                    existing_func_names = []
```
Comment (moranbental) [inline mlrun/projects/project.py]: You don’t need to initialize the list - it is already done later.

File: mlrun/projects/project.py [hunk 2]
```diff
@@ -2424,9 +2424,18 @@ def _instantiate_function(
...
+                    existing_func_defs = self.list_functions(name=func)
+                    if existing_func_defs:
+                        existing_func_names = [existing_func_def.metadata.name for existing_func_def in
+                                               existing_func_defs]
+                    if not existing_func_defs or func not in existing_func_names:
```
Comment (moranbental) [inline mlrun/projects/project.py]:
```suggestion
                    existing_func_names = [existing_func_def.metadata.name for existing_func_def in existing_func_defs]
                    if func not in existing_func_names:
```

File: mlrun/projects/project.py [hunk 3]
```diff
@@ -2424,9 +2424,18 @@ def _instantiate_function(
             )
 
         if isinstance(func, str):
-            # in hub or db functions name defaults to the function name
-            if not name and not (func.startswith("db://") or func.startswith("hub://")):
-                raise ValueError("Function name must be specified")
+            # In hub or db functions name defaults to the function name
+            # If name is specified, we try to create a new function with this name.
```
Comment (moranbental) [inline mlrun/projects/project.py]: I suggest moving this comment `# If name is specified, we try to create a new function with this name.` to before line 3439, where the function is actually created.

File: mlrun/projects/project.py [hunk 4]
```diff
@@ -2424,9 +2424,18 @@ def _instantiate_function(
             )
 
         if isinstance(func, str):
-            # in hub or db functions name defaults to the function name
-            if not name and not (func.startswith("db://") or func.startswith("hub://")):
-                raise ValueError("Function name must be specified")
```
Comment (moranbental) [inline mlrun/projects/project.py]: What about raising this error: `raise ValueError("Function name must be specified")`

**Review by moranbental — CHANGES_REQUESTED** [review]
Looks good! Some comments below. 
Also, please add unit tests.

**Comment by TomerShor** [comment]
@adamdelman make sure to run `make fmt lint` before you commit your changes

**Pull request closed without merging**